### ISSUE TITLE
Beer Factory Managers

### ISSUE THREAD

**Issue by wol-soft** [wol-soft/beerplop#104]
Beer Factory Managers can be used to increase the production of Beer Factory factories and factory extensions. Managers consume food of a specific level to perform their work. The basic food is produced by new factories. Higher level food can be produced in a new factory, the *restaurant*.

## Additional factories (separated in the factory view)
- Crop field -> produces crops
	- factory extension: Bakery -> produces bread
- Orchard -> produces fruits
	- factory extension: Grandma's House -> produces jam
- Greenhouse -> produces vegetables
	- factory extension: garden shed -> produces mixed salad
- Fisherman's Hut -> produces fish
	- factory extension: Smokehouse -> produces smoked fish
- Cattle farm -> produces meat
	- factory extension: Slaughterhouse -> produces steak
- Restaurant

### Food production via new factories

* Each food creating factory produces a Level I food
	- Unlocked at
		- Crop field: 10 academies (together with the general availability of managers and the academy training factory extension)
		- Orchard: 5 Crop Fields
		- Greenhouse: 5 Orchards
		- Fisherman: 5 Greenhouses
		- Cattle Farm: 5 Fisherman's Hut
* The factory extensions produce Level II food
	- One level II food produced with a factory extension consumes 3 base foods
	- Factory extensions unlock after owning 5 of the corresponding factories
* Each food creating factory has an "Distillery"-Upgrade which unlocks an additional Beer Blender Bar ingredient. The distillery upgrade unlocks after owning 10 of the corresponding factories.

### Food production via the Restaurant

The restaurant is required to craft food > Level II
- The amount of constructed restaurants increases the amount of meals which can be produced in a single iteration
- The player can create meals
	- A single meal contains base foods from which it is cooked
	- Define a cooking amount (the sum of all cooking amounts must not exceed the total meal limit)
	- The sum of the base food level defines the resulting food level (eg. a recipe from Bread, Fruits and a Fish will result in a level IV food)
	- Each base food can be used only once per meal (so no recipe like Fruits, Fruits, Fish; but Fruits, Jam, Fish is possible)
	- Meals can be named
	- Each cooked meal consumes 1 base food of each base food used in the recipe
- Upgrades of the restaurant
	- increase the amount of base foods which can be used in a single meal
	- increase the amount of different meals which can be defined

## Basic manager functionality

- Each factory can be boosted by multiple managers
- Managers can be hired. After hiring a new manager the manager will be trained at the academy
	- project based education of managers via a new factory extension at the academy
		- Only one manager at a time can be trained
		- Multiple managers can be queued for training
		- Queue Management: Rearrange order/remove managers
		- TODO: should it be possible to proxy the extension to parallelize the trainings?
	- consumes knowledge
	- while a manager waits for his training/is trained he's inactive
- Managers have different skills/efficiencies based on their level
	- The level defines which food the manager consumes
	- If the manager doesn't get food of his level (or a higher level) he's inactive
		- He will consume food of a lower level to avoid starving
		- If the manager is inactive for X iterations he will level down
- Each manager starts at level I and must be upgraded individually
	- Upgrades are only available if enough food of the next level is produced (food of an even higher level may be consumed, eg. a manager which is upgraded from level II to level III may consume a level IV meal from the restaurant)
	- Upgrades require an additional education training in the academy
		- training cost (amount of required knowledge) increases with the level
		- during a training the manager is inactive
		- while a manager waits for his training he's still active
- Managers can be named
		
## Effects of managers

- Managers for Lumberjack, Quarry, ... boost the production of the factory
	- Higher levels increase the boost or additionally boost also the factory extensions
	- Managers may also be applied to a single factory extension boosting the production of the extension
		- Two ways of boosting the production:
			- Increase produced amount (for direct production, eg. Charcoal)
			- Reduce the amount of required materials for a single production step (for project production, eg. Base Plates)
- Managers for Storage Places extend the storage capacity
	- Higher levels either increase the general storage capacity or the factory extension storage capacity
- Managers for Transport Systems increase the transport capacity
- Managers for Worker Lodges increase the production and transport capacity multiplier
- Managers for Builder Yards increase the price reduction multipliers
- Managers for Trading Posts increase the amount of tradable items
- Managers for Back Rooms increase the influence of Lobbyists
	- Display the influence of lobbyists in the Back Room details
- Managers for Restaurants increase the amount of meals which can be produced in a single iteration
- Queue Control and Engineers can't be affected by managers

## Additional UI change:

Split current materials view

- Base materials
- Crafted materials
- Food

Each section shall be collapsible

**Comment by wol-soft** [comment]
Partial development merged into master to pull bugfixes from the development branch into master
